lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestRestBasedAccessControlConfig.java

```diff
@@ -0,0 +1,88 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.ConfigurationException;
+import io.airlift.configuration.ConfigurationFactory;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
+import static io.airlift.configuration.testing.ConfigAssertions.assertRecordedDefaults;
+import static io.airlift.configuration.testing.ConfigAssertions.recordDefaults;
+import static io.trino.plugin.base.security.RestBasedAccessControlConfig.SECURITY_CONFIG_URL;
+import static io.trino.plugin.base.security.RestBasedAccessControlConfig.SECURITY_JSON_POINTER;
+import static io.trino.plugin.base.security.RestBasedAccessControlConfig.SECURITY_REST_REFRESH_PERIOD;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestRestBasedAccessControlConfig
+{
+    @Test
+    public void testDefaults()
+    {
+        assertRecordedDefaults(recordDefaults(RestBasedAccessControlConfig.class)
+                .setRestUrl(null)
+                .setJsonPointer("")
+                .setRefreshPeriod(null));
+    }
+
+    @Test
+    public void testExplicitPropertyMappingsWithUri()
+    {
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(SECURITY_CONFIG_URL, "http://test:1234/example")
+                .put(SECURITY_JSON_POINTER, "/data")
+                .put(SECURITY_REST_REFRESH_PERIOD, "1s")
+                .buildOrThrow();
+
+        RestBasedAccessControlConfig expected = (RestBasedAccessControlConfig) new RestBasedAccessControlConfig()
+                .setRestUrl("http://test:1234/example")
+                .setJsonPointer("/data")
+                .setRefreshPeriod(new Duration(1, TimeUnit.SECONDS));
+
+        assertFullMapping(properties, expected);
+    }
+
+    @Test
+    public void testValidation()
+            throws IOException
+    {
+        Path securityConfigFile = Files.createTempFile(null, null);
+
+        assertThatThrownBy(() -> newInstance(ImmutableMap.of(SECURITY_REST_REFRESH_PERIOD, "1ms")))
+                .isInstanceOf(ConfigurationException.class)
+                .hasMessageContaining("security.config-url: may not be null ");
+
+        assertThatThrownBy(() -> newInstance(ImmutableMap.of(
+                SECURITY_CONFIG_URL, securityConfigFile.toString(),
+                SECURITY_REST_REFRESH_PERIOD, "1us")))
+                .isInstanceOf(ConfigurationException.class)
+                .hasMessageContaining("Invalid configuration property security.refresh-period");
+
+        newInstance(ImmutableMap.of(SECURITY_CONFIG_URL, securityConfigFile.toString()));
```

Comment:
We can use `assertFailsValidation` for a specific field.

Reply:
not sure it can be used here as we instantiate an object via ConfigurationFactory and all validation failures are wrapped com.google.inject.ConfigurationException

Reply:
We use them for config validations - in `TestLdapConfig`

Reply:
I saw it - but you just use constructor to create an object and later it is validated inside assertFailsValidation. Here we use configurationFactory.build(FileBasedAccessControlConfig.class); And it triggers validation and just throws ConfigurationException in case any validation failure

Reply:
Asserting based on `configurationFactory#build` introduces a flakiness  as we don't have a control over the order in which the validation would be applied. We need to isolate the test case for each validation which is implemented in `assertFailsValidation`

Reply:
To correct my previous statement - I'm not sure if it is flaky - but the validation assertion has been evolved from building a config object which validates as a part of creation to this `assertFailsValidation` which allows us to check each validation method in an elegant manner.